Algorithm/QCAlgorithm.cs

```diff
@@ -1809,5 +1810,21 @@ public void SetStatus(AlgorithmStatus status)
         {
             Status = status;
         }
+
+        /// <summary>
+        /// Downloads the requested resource as a <see cref="string"/>.
+        /// The resource to download is specified as a <see cref="string"/> containing the URI.
+        /// </summary>
+        /// <param name="address">A string containing the URI to download</param>
+        /// <param name="userName">The user name associated with the credentials</param>
+        /// <param name="password">The password for the user name associated with the credentials</param>
+        /// <returns>The requested resource as a <see cref="string"/></returns>
+        public string Fetch(string address, string userName = null, string password = null)
```

Comment:
Some web services don't use network credentials, but instead use authorization headers. Do we want/need to support such a case?